src/psim/truth/hill_frame.cpp

```diff
@@ -30,56 +30,51 @@
 
 #include <gnc/utilities.hpp>
 
-#include <lin/core.hpp>
-
 namespace psim {
 
 HillFrameEci::HillFrameEci(RandomsGenerator &randoms,
-    Configuration const &config, std::string const &leader,
-    std::string const &follower)
-  : Super(randoms, config, leader, follower, "eci") {}
+    Configuration const &config, std::string const &satellite,
+    std::string const &other)
+  : Super(randoms, config, satellite, other, "eci") {}
+
+Vector4 HillFrameEci::truth_satellite_hill_q_hill_frame() const {
+  auto const &satellite_r = truth_satellite_orbit_r_frame->get();
+  auto const &satellite_v = truth_satellite_orbit_v_frame->get();
 
-Vector4 HillFrameEci::truth_leader_q_hill_frame() const {
-  auto const &leader_r = this->truth_leader_orbit_r_frame->get();
-  auto const &leader_v = this->truth_leader_orbit_v_frame->get();
+  Matrix<3, 3> Q_hill_frame;
+  gnc::utl::dcm(Q_hill_frame, satellite_r, satellite_v);
 
-  lin::Vector<Real, 4> q;
-  lin::Matrix<Real, 3, 3> Q;
-  gnc::utl::dcm(Q, leader_r, leader_v); // Q = DCM_hill_eci
-  gnc::utl::dcm_to_quat(Q, q);          // q = q_hill_eci
-  gnc::utl::quat_conj(q);               // q = q_eci_hill
-  return q;
+  Vector<4> q_hill_frame;
+  gnc::utl::dcm_to_quat(Q_hill_frame, q_hill_frame);
+  return q_hill_frame;
```

Comment:
I'm not sure why there used to be the `gnc::utl::quat_conj(q);` line in there... We are interested in `q_hill_eci` which converts from ECI to the HILL frame. This transform will be used later to rotate into the HILL frame.